### PR TITLE
Add Type Lookup Service example using DynamicTypes

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: resources/dds/TypeLookupService/types/complete/Complete.h [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+
+private:
+
+    uint32_t m_index;
+    Point m_main_point;
+    std::vector<Point> m_internal_data;
+    std::array<Message, 2> m_messages;
+};
+
+#endif // _FAST_DDS_GENERATED_COMPLETE_H_
```
Comment (irenebm) [inline resources/dds/TypeLookupService/types/complete/Complete.h:994]: EOL

File: resources/dds/TypeLookupService/types/complete/CompletePubSubTypes.h [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        return false;
+    }
+
+#endif  // TOPIC_DATA_TYPE_API_HAS_CONSTRUCT_SAMPLE
+
+    MD5 m_md5;
+    unsigned char* m_keyBuffer;
+};
+
+#endif // _FAST_DDS_GENERATED_COMPLETE_PUBSUBTYPES_H_
```
Comment (irenebm) [inline resources/dds/TypeLookupService/types/complete/CompletePubSubTypes.h:357]: EOF

File: resources/dds/TypeLookupService/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+* How to create a generic Subscriber able to subscribe to any incoming data type using DynamicTypes.
+
+
+## Building the example
+
+To build the example, install Fast DDS following one of the [installation methods](https://fast-dds.docs.eprosima.com/en/latest/installation/binaries/binaries_linux.html) described in its documentation, or by [installing DDS Recorder](https://dds-recorder.readthedocs.io/en/latest/rst/installation/windows.html) together with eProsima dependencies, i.e. Fast DDS, Fast CDR and DDS Router.
+
+Once Fast DDS is installed, source Fast DDS or make the libraries accessible to be used by CMake.
+
+Finally, go to this same folder and run:
```
Comment (irenebm) [inline resources/dds/TypeLookupService/README.md:23]:
```suggestion
Finally, go to this folder and run:
```

File: resources/dds/TypeLookupService/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+```bash
+mkdir build
+cd build
+cmake .. && make -j8
+```
+
+## Recording samples with DDS Recorder
+
+1.  Open to terminals
```
Comment (irenebm) [inline resources/dds/TypeLookupService/README.md:33]:
```suggestion
1.  Open two terminals
```

File: resources/dds/TypeLookupService/arg_configuration.h [hunk 5]
```diff
@@ -0,0 +1,222 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (irenebm) [inline resources/dds/TypeLookupService/arg_configuration.h:1]:
```suggestion
// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: resources/dds/TypeLookupService/main.cpp [hunk 6]
```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (irenebm) [inline resources/dds/TypeLookupService/main.cpp:1]:
```suggestion
// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: resources/dds/TypeLookupService/README.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+General options:
+  -h, --help                        Produce help message.
+  -e, --entity <dds_entity>         DDS Entity type (Default: publisher).
+                                    Allowed options:
+                                    • publisher -> Run a DDS Publisher.
+                                    • subscriber -> Run a DDS Subscriber.
+
+Publisher options:
+  -t, --topic <topic_name>          Topic name (Default: DDSTopic).
```
Comment (irenebm) [inline resources/dds/TypeLookupService/README.md:78]:
```suggestion
  -t, --topic <topic_name>          Topic name (Default: /dds/topic).
```

File: resources/dds/TypeLookupService/TypeLookupServiceSubscriber.h [hunk 8]
```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (irenebm) [inline resources/dds/TypeLookupService/TypeLookupServiceSubscriber.h:1]:
```suggestion
// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: resources/dds/TypeLookupService/TypeLookupServiceSubscriber.cpp [hunk 9]
```diff
@@ -0,0 +1,318 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (irenebm) [inline resources/dds/TypeLookupService/TypeLookupServiceSubscriber.cpp:1]:
```suggestion
// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+    // Create the DDS Topic
+    topic_ = participant_->create_topic(topic_name_, data_type_name_, TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        throw std::runtime_error("Error creating topic");
+    }
+
+    ///////////////////////
+    // Create the DDS Topic
```
Comment (irenebm) [inline resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp:116]:
```suggestion
    // Create the DDS Data Writer
```

**Review by irenebm — CHANGES_REQUESTED** [review]

File: resources/dds/TypeLookupService/types/complete/Complete.h (hunk 1 above)
Comment (rsanchez15) [reply resources/dds/TypeLookupService/types/complete/Complete.h:994]: We do not modify autogenerated files.

**Review by irenebm — APPROVED** [review]

**Comment by irenebm** [comment]
LGTM

**Pull request merged**